experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -353,6 +354,171 @@ describe("Editing", () => {
 
 			expectJsonTree([tree, tree1, tree2], [{ foo: [{}, "b"] }]);
 		});
+
+		describe.skip("Exhaustive removal tests", () => {
+			// Toggle the constant below to run each scenario as a separate test.
+			// This is useful to debug a specific scenario but makes CI and the test browser slower.
+			// Note that if the numbers of nodes and peers are too high (more than 3 nodes and 3 peers),
+			// then the number of scenarios overwhelms the test browser.
+			// Should be committed with the constant set to false.
+			const individualTests = false;
+			const nbNodes = 3;
+			const nbPeers = 3;
+			const testRemoveRevive = true;
+			const testMoveReturn = true;
+			assert(testRemoveRevive || testMoveReturn, "No scenarios to run");
+
+			const [outerFixture, innerFixture] = individualTests
+				? [describe, it]
+				: [it, (title: string, fn: () => void) => fn()];
+
+			type ScenarioStep =
+				| { readonly removeIdx: number; readonly peer: number }
+				| { readonly peer: number };
+
+			/**
+			 * Generates all permutations for `nbNodes` and `nbPeers` such that:
+			 * - Each node is removed exactly once.
+			 * - Each removal is undone by the peer that removed it.
+			 * The order of removals and undos is unique when considering which peer does what.
+			 * This does mean that this function produces symmetrical scenarios such as:
+			 * - D(i:0 p:0) D(i:1 p:1) U(1) U(0)
+			 * - D(i:0 p:1) D(i:1 p:0) U(0) U(1)
+			 * This is taken advantage of to test different network conditions (see {@link runScenario}).
+			 */
+			function buildScenarios(): Generator<readonly ScenarioStep[]> {
+				interface ScenarioBuilderState {
+					removed: boolean[];
+					peerQueue: number[];
```

Comment:
it's not clear to me why this is called `peerQueue`. But I probably care more that it's doc'd rather than needing to be inferred based on usage. seems like this is the field's semantics, but feel free to write something a bit different:


```suggestion
					/**
					 * peerQueue[i] tracks the number of outstanding delete operations that the `i`th peer has yet to undo.
					 */
					peerQueue: number[];
```